packages/wpcom-proxy-request/src/index.js

```diff
@@ -220,6 +220,22 @@ export function requestAllBlogsAccess() {
  */
 
 function submitRequest( params ) {
+	// Sometimes the `iframe.contentWindow` is `null` even though the `iframe` has been correctly
+	// loaded. Can happen when some other buggy script removes it from the document.
+	if ( ! iframe.contentWindow ) {
+		debug( 'proxy iframe is not present in the document' );
+		// Look up the issuing XHR request and make it fail
+		const id = params.callback;
+		const xhr = requests[ id ];
+		delete requests[ id ];
+		reject(
+			xhr,
+			new WPError( { status_code: 500, error_description: 'proxy iframe element is not loaded' } ),
```

Comment:
Furthermore, I wonder if `500` is the right error type. This appears more like a client-side issue and not a server-side issue, so I expected a 4xx status code, like `406` or `405`. WDYT?

Reply:
A client-side issue, reported as a 4xx error, means that there is something wrong with the request. Bad URL, missing information, some part is too large... But here the request is perfectly fine, we're just unable to process it. That's why I think 5xx is better. There are 5xx errors like 502 Bad Gateway or 504 Gateway Timeout that talk about a "gateway" going wrong. Maybe we could use 502 because that's exactly what we have here, a broken gateway (the proxy frame). But there's very little practical difference between 500 and 502.

Reply:
The thing is we're unable to process it on the client, not on the server, which is why I fail to see why we'd want to use a server error code. Anyway, I'm happy to leave the final decision to you.

Reply:
In our case, the client is the app code that calls `wp.req.get()`. If there was some problem with the request's method, URL or headers, that would be a reason for a 4xx error.

The proxy iframe is not a client, it's a proxy. It runs in the browser, in the same tab (but different window and document) as the client code, but it's not _the_ client. It processes the requests originated in the client.

> I expected a 4xx status code, like 406 or 405.

Even if I wanted to use a 4xx code, I don't see any good one. 400 means "bad request", but our request is good. 406 means "not acceptable", i.e., the constraints specified in `Accept-*` headers can't be satisfied. Not our case. 405 means "method not allowed", like when the request is `PUT` but the server supports only `POST`. Also irrelevant.

On the other hand, there are good 5xx errors about gateways, like 502.

Reply:
Your 502 rationale makes sense 👍